plugins/modules/sonic_image_management.py

```diff
@@ -0,0 +1,429 @@
+
+    else:
+        method = 'POST'
+        payload = {'openconfig-image-management:input': {}}
+        if params['category'] == 'image':
+            if params['command'] == 'install':
+                payload['openconfig-image-management:input'] = {'image-name': params['path']}
+            elif (params['command'] == 'remove' and params['name'] != 'all') or params['command'] == 'set-default':
+                payload['openconfig-image-management:input'] = {'image-name': params['name']}
+        elif params['category'] == 'patch':
+            if params['command'] == 'install':
+                payload['openconfig-image-management:input'] = {'patch-name': params['path'], 'skip-image-check': ''}
+            elif params['command'] == 'rollback':
+                payload['openconfig-image-management:input'] = {'patch-name': params['name']}
+        elif params['category'] == 'firmware':
+            if params['command'] == 'install':
+                payload['openconfig-image-management:input'] = {'fwpkg-name': params['path']}
+
+        request = [{'path': path, 'method': method, 'data': payload}]
+        try:
+            response = edit_config(module, to_request(module, request))
+        except ConnectionError as exc:
+            module.fail_json(msg=str(exc), code=exc.code)
+
+        status = ''
+        response = response[0][1].get('openconfig-image-management:output')
+        if response:
+            if response['status'] != 0:
+                status = response['status-detail']
+            else:
+                status = command_map[params['category']][params['command']].get('status', response['status-detail'])
+
+        result['status'] = status
+
+
+def main():
+    """
+    Main entry point for module execution
+    """
+    argument_spec = {
+        'image': {
+            'type': 'dict',
+            'options': {
+                'command': {
+                    'type': 'str',
+                    'required': True,
+                    'choices': ['install', 'cancel', 'remove', 'set-default', 'get-list', 'get-status']
+                },
+                'name': {'type': 'str'},
+                'path': {'type': 'str'}
+            }
+        },
+        'patch': {
+            'type': 'dict',
+            'options': {
+                'command': {
+                    'type': 'str',
+                    'required': True,
+                    'choices': ['install', 'rollback', 'get-history', 'get-list', 'get-status']
+                },
+                'name': {'type': 'str'},
+                'path': {'type': 'str'}
+            }
+        },
+        'firmware': {
+            'type': 'dict',
+            'options': {
+                'command': {
+                    'type': 'str',
+                    'required': True,
+                    'choices': ['install', 'cancel', 'get-list', 'get-status']
+                },
+                'path': {'type': 'str'}
+            }
+        }
+    }
```

Comment:
Any reason for the argument spec is not put into a argspec file under argspec directory?

Reply:
This module is similar to 'sonic_command', 'sonic_config' modules and hence the module implementation is done in the module file itself with no separate argspec, facts and config files.

Reply:
Thanks.